mmv1/third_party/cai2hcl/common/uber_converter.go

```diff
@@ -0,0 +1,46 @@
+package common
+
+import (
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v5/caiasset"
+)
+
+// Converter which aggregates all service-specific converters in the same interface.
+type UberConverter struct {
```

Comment:
What is the purpose of this refactor? Why is it necessary for the upcoming changes?

Reply:
The main reason was to incorporate new converter matching strategy - "by asset selfLink RegExp". This is needed to map ComputeForwardingRule asset type to either "compute_forwarding_rule" or "global_compute_forwarding_rule" converter based on a selfLink regexp. In Compute API both of resources have the same type - this is a common practice in Compute.

This new strategy will add 3rd property in the converter_map.go "interface", while there is no such interface in code. So the main purpose was to create a single interface for service-level converters aggregator.

Reply:
Here is an example of a matcher implementation: https://github.com/amirkaromashkin/magic-modules/blob/feature/cai2hcl_provider_generators/mmv1/third_party/cai2hcl/services/compute/converter.go